orchagent/lagids.lua

```diff
@@ -0,0 +1,90 @@
+-- KEYS - None
+-- ARGV[1] - operation (add/del/get)
+-- ARGV[2] - lag name
+-- ARGV[3] - current lag id (for "add" operation only)
+
+-- return lagid if success for "add"/"del"
+-- return 0 if lag not exists for "del"
+-- return -1 if lag table full for "add"
+-- return -2 if log does exist for "get"
+-- return -3 if invalid operation
+
+local op = ARGV[1]
+local pcname = ARGV[2]
+
+local lagid_start = tonumber(redis.call("get", "SYSTEM_LAG_ID_START"))
+local lagid_end = tonumber(redis.call("get", "SYSTEM_LAG_ID_END"))
+
+if op == "add" then
+
+    local plagid = tonumber(ARGV[3])
+
+    local dblagid = redis.call("hget", "SYSTEM_LAG_ID_TABLE", pcname)
+
+    if dblagid then
+        dblagid = tonumber(dblagid)
+        if plagid == 0 then
+            -- no lagid propsed. Return the existing lagid
+            redis.call("sadd", "SYSTEM_LAG_ID_SET", tostring(dblagid))
+            return dblagid
+        end
+    end
+
+    -- lagid allocation request with a lagid proposal
+    if plagid >= lagid_start and plagid <= lagid_end then
+        if plagid == dblagid then
+            -- proposed lagid is same as the lagid in database
+            redis.call("sadd", "SYSTEM_LAG_ID_SET", tostring(plagid))
+            return plagid
+        end
+        -- proposed lag id is different than that in database OR
+        -- the portchannel does not exist in the database
+        -- If proposed lagid is available, return the same proposed lag id
+        if redis.call("sismember", "SYSTEM_LAG_ID_SET", tostring(plagid)) == 0 then
+            redis.call("sadd", "SYSTEM_LAG_ID_SET", tostring(plagid))
+            redis.call("srem", "SYSTEM_LAG_ID_SET", tostring(dblagid))
```

Comment:
dblagid could be invalid, incase the portchannel doesn't exist in DB. Can you confirm this call don't error out.

Reply:
This will not error out, instead a new lag id will be allocated.
If the portchannel does not exist, dblagid will be "".  In that case, if the proposed lagid is available, we allocate the proposed lag id. If the proposed lad is not available, we continue and allocate new lagid.